File: VB/FilteringRepeatRows/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace FilteringRepeatRows
	Partial Public Class Form1
		Inherits Form
		Private filterFields(1) As List(Of String)
		Private helper As FilterHelper

		Public Sub New()
			InitializeComponent()
			Dim fields As New List(Of String)()
			fields.Add("Name")
			fields.Add("Company")
			fields.Add("Address")
			filterFields(0) = Nothing
			filterFields(1) = fields
			helper = New FilterHelper(gridView1)
			radioGroup1.SelectedIndex = 0
			gridControl1.DataSource = CreateTable(20)
		End Sub

		Private Function CreateTable(ByVal rowCount As Integer) As DataTable
			Dim table As New DataTable()
			table.Columns.Add("Name")
			table.Columns.Add("Company")
			table.Columns.Add("Age", GetType(Integer))
			table.Columns.Add("Address")
			Dim repeatCount As Integer = 3
			For i As Integer = 0 To rowCount - 1
				For j As Integer = 0 To repeatCount - 1
					table.Rows.Add(New Object() { String.Format("Name{0}", i), String.Format("Company{0}", i), 25 + i * repeatCount + j, String.Format("Address{0}", i) })
				Next j
			Next i
			Return table
		End Function

		Private Sub radioGroup1_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles radioGroup1.SelectedIndexChanged
			helper.Fields = filterFields(radioGroup1.SelectedIndex)
		End Sub
	End Class
End Namespace

File: CS/FilteringRepeatRows/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace FilteringRepeatRows
{
    public partial class Form1 : Form
    {
        List<string>[] filterFields = new List<string>[2];
        FilterHelper helper;

        public Form1()
        {
            InitializeComponent();
            List<string> fields = new List<string>();
            fields.Add("Name");
            fields.Add("Company");
            fields.Add("Address");
            filterFields[0] = null;
            filterFields[1] = fields;
            helper = new FilterHelper(gridView1);
            radioGroup1.SelectedIndex = 0;
            gridControl1.DataSource = CreateTable(20);
        }

        DataTable CreateTable(int rowCount)
        {
            DataTable table = new DataTable();
            table.Columns.Add("Name");
            table.Columns.Add("Company");
            table.Columns.Add("Age", typeof(int));
            table.Columns.Add("Address");
            int repeatCount = 3;
            for (int i = 0; i < rowCount; i++)
            {
                for (int j = 0; j < repeatCount; j++)
                {
                    table.Rows.Add(new object[] { String.Format("Name{0}", i), String.Format("Company{0}", i), 25 + i * repeatCount + j, String.Format("Address{0}", i) });
                }
            }
            return table;
        }

        private void radioGroup1_SelectedIndexChanged(object sender, EventArgs e)
        {
            helper.Fields = filterFields[radioGroup1.SelectedIndex];
        }
    }
}
